File: CS/SilverlightApplication1/DinamicDictionary.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using System.Dynamic;
using System.Collections.Generic;

namespace SilverlightApplication1 {
    public class DynamicDictionary : DynamicObject {
        Dictionary<string, object> dictionary
            = new Dictionary<string, object>();

        public int Count {
            get {
                return dictionary.Count;
            }
        }
        public override bool TryGetMember(
            GetMemberBinder binder, out object result) {
            string name = binder.Name.ToLower();
            return dictionary.TryGetValue(name, out result);
        }

        public override bool TrySetMember(
            SetMemberBinder binder, object value) {
            dictionary[binder.Name.ToLower()] = value;
            return true;
        }
    }
}

File: VB/SilverlightApplication1/DinamicDictionary.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Net
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Documents
Imports System.Windows.Ink
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes
Imports System.Dynamic
Imports System.Collections.Generic

Namespace SilverlightApplication1
	Public Class DynamicDictionary
		Inherits DynamicObject
		Private dictionary As New Dictionary(Of String, Object)()

		Public ReadOnly Property Count() As Integer
			Get
				Return dictionary.Count
			End Get
		End Property
		Public Overrides Function TryGetMember(ByVal binder As GetMemberBinder, <System.Runtime.InteropServices.Out()> ByRef result As Object) As Boolean
			Dim name As String = binder.Name.ToLower()
			Return dictionary.TryGetValue(name, result)
		End Function

		Public Overrides Function TrySetMember(ByVal binder As SetMemberBinder, ByVal value As Object) As Boolean
			dictionary(binder.Name.ToLower()) = value
			Return True
		End Function
	End Class
End Namespace
